Sources/Bot/Services/DispatchService.swift

```diff
@@ -0,0 +1,208 @@
+import Foundation
+import Result
+import ReactiveSwift
+import ReactiveFeedback
+
+/// Orchestrates multiple merge services, one per each target branch of PRs enqueued for integration
+public final class DispatchService {
+    private let integrationLabel: PullRequest.Label
+    private let topPriorityLabels: [PullRequest.Label]
+    private let requiresAllStatusChecks: Bool
+    private let statusChecksTimeout: TimeInterval
+
+    private let logger: LoggerProtocol
+    private let gitHubAPI: GitHubAPIProtocol
+    private let scheduler: DateScheduler
+
+    /// Merge services per target branch
+    private var mergeServices: [String: MergeService]
```

Comment:
This needs to be thread-safe so we need to wrap it using `Atomic` and using the proper methods to access and mutate it, we may have PRs coming while one of the merge services changes to a `idle` state.

Reply:
@dmcrodrigues Should be fixed in 700d62d8e9d2d67c9606d76ae52c218a23e59275 but another pair of eyes to double-check is welcome